src/app/Library/CrudPanel/Traits/FieldsProtectedMethods.php

```diff
@@ -169,20 +169,25 @@ protected function makeSureFieldHasRelationshipData($field)
         return $field;
     }
 
-    protected function makeSureFieldNameMatchesRelation($field)
+    protected function overwriteFieldNameFromEntity($field)
     {
-        switch ($field['relation_type']) {
-            case 'BelongsTo':
-                if (count(explode('.', $field['entity'])) == count(explode('.', $this->getOnlyRelationEntity($field)))) {
-                    $field['name'] = implode('.', array_slice(explode('.', $field['entity']), 0, -1));
-                    $relation = $this->getRelationInstance($field);
-                    if (! empty($field['name'])) {
-                        $field['name'] .= '.';
-                    }
-                    $field['name'] .= $relation->getForeignKeyName();
-                }
-
-        break;
+        // if the entity doesn't have a dot, it means we don't need to overwrite the name
+        if (! Str::contains($field['entity'], '.')) {
+            return $field;
+        }
+
+        // only 1-1 relationships are supported, if it's anything else, abort
+        if ($field['relation_type'] != 'BelongsTo') {
+            return $field;
+        }
```

Comment:
Same thing as before, but I just refactored it from a `switch` to an `if`. Since it only has one branch.